libcalico-go/lib/backend/k8s/conversion/conversion.go

```diff
@@ -417,6 +417,12 @@ func k8sANPIngressRuleToCalico(rule adminpolicy.AdminNetworkPolicyIngressRule) (
 			break
 		}
 
+		// Named ports do not have protocol
+		if protocol == nil {
+			protocolPorts[""] = []numorstring.Port{*calicoPort}
+			continue
+		}
+
```

Comment:
@fasaxc Originally I handled it like this, but if a rule combines named ports with other ports, this way a separate rule would generate for named ports.

Reply:
I _think_ that's the correct interpretation.  Question is: does the dataplane handle that correctly (I'm not sure if you can match on ports without specifying a protocol so possible we fail or default it to TCP or something)?

Reply:
The conformance tests are passing, and those tests include named ports with both udp and tcp. From what I've seen, it seems the protocol does not matter for named ports. The underlying struct does not include a protocol value.

Reply:
It looks we default to TCP here: https://github.com/projectcalico/calico/blob/d377c71518daf297fa7eb90711d7cfe2a8f92999/felix/calc/rule_scanner.go#L390

Also the conformance tests only include tcp protocols case. I think we should default to tcp here, and explicitly state that in our docs. WDYT?